### DISCUSSION TITLE
CameraView Scaling - Android

### DISCUSSION THREAD

**Discussion by Jake-Derrick** [CommunityToolkit/Maui#1997] [General]
I noticed that for the Android camera `PreviewView`, the ScaleType was set to `ScaleType.FitCenter`. Was there a reason this was picked over the default value of `ScaleType.FillCenter`? I feel `ScaleType.FillCenter` might be more expected and seems to align better with the ios view too.

| Android FitCenter (default) | Android FillCenter | iOS (default) |
| --- | --- | --- |
| ![Android FitCenter](https://github.com/CommunityToolkit/Maui/assets/60721064/d6126d5e-7754-4705-879e-e8c9ba2a0d97) | ![Android FillCenter](https://github.com/CommunityToolkit/Maui/assets/60721064/514934c3-514b-42e5-9098-a9bd77069727) | ![iOS default](https://github.com/CommunityToolkit/Maui/assets/60721064/c145b970-89fc-4ffe-9849-bb26eec721f1) |

**Comment by ne0rrmatrix** [comment]
Fill Center looks better and matches the same behavior as iOS. That looks like a good idea to implement.

**Comment by bijington** [comment]
If you capture an image what is the result? I can see that both options in the preview isn't showing all of the same detail - this might be down to a slightly different camera angle.

I think the best option is for the preview to most accurately match the captured image and not other platforms. The being said I think we should probably let the developer decide and make this configurable

**Reply by Jake-Derrick** [reply]
That's a good point. Both the Android with FillCenter and iOS will include extra detail in the captured image that isn't shown in the PreviewView. I think for Android [ViewPort](https://developer.android.com/reference/androidx/camera/core/ViewPort) can be used align the PreviewView and the captured image, but I haven't used it yet to see how it works.

At any rate, I agree making it configurable would probably be the best way to go.
